app/models/user/block_domain.rb

```diff
@@ -0,0 +1,6 @@
+class User::BlockDomain < ApplicationRecord
+  def self.blocked?(user)
+    domain = Mail::Address.new(user.email).domain
```

Comment:
Is it possible for a user to not have an email?

Reply:
No.